### PR TITLE
[yang]YANG model for policer table

### PR DESCRIPTION
**Pull request by dgsudharsan**
#### Why I did it
Added yang model for policer table
Fixes https://github.com/Azure/sonic-buildimage/issues/9742 and https://github.com/Azure/sonic-buildimage/issues/9743
#### How I did it
Creating yang model for policer

#### How to verify it
Added UT to verify the yang model

The configuration schema for policer is added in the pull request https://github.com/Azure/sonic-swss/pull/2144

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: src/sonic-yang-models/tests/yang_model_tests/tests/policer.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+    "POLICER_TABLE": {
+    "desc": "Configure policer with all fields."
```
Comment (qiluo-msft) [inline src/sonic-yang-models/tests/yang_model_tests/tests/policer.json:3]: >"desc" [](http://example.com/codeflow?start=4&length=6)

Could you reformat this json file?

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+					description "Policer color Source";
+				} 
+
+				leaf cir {
+					type uint64;
+					default 0;
+					description
+						"Committed information rate for the dual-rate token
+						bucket policer.  This value represents the rate at which
+						tokens are added to the primary bucket.";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:57]: what's the unit? kbps?

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "cbs must be greater than or equal to cir";
+					}
+					type uint64;
+					default 0;
+					description
+						"Committed burst size for the dual-rate token bucket
+						policer.  This value represents the depth of the token
+						bucket.";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:72]: what's the unit? bytes?

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+						"Committed information rate for the dual-rate token
+						bucket policer.  This value represents the rate at which
+						tokens are added to the primary bucket.";
+				}
+
+				leaf cbs {
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "cbs can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:64]: Please refer to https://datatracker.ietf.org/doc/html/rfc4115.
The CBS and EBS are measured in bytes and must configure to be greater than the expected maximum length of the incoming PDU.  The CIR and EIR are both measured in bits/s.
I think we need to convert unit for comparing.

**Comment by dgsudharsan** [comment]
@ganglyu  Please refer to copp yang table. I have reused the same definitions used there. Basically copp internally uses policer and both should be aligned in yang definitions and so I defined as in https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-copp.yang

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 (hunk 2 above)
Comment (dgsudharsan) [reply src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:57]: It depends on the meter type defined. The meter type can be bytes or packets. This field by itself has no units. Please refer to https://github.com/opencomputeproject/SAI/blob/16c11ee7062e630f1eaf4bb3e44d31ab58eafa5b/inc/saipolicer.h#L123

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 (hunk 3 above)
Comment (dgsudharsan) [reply src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:72]: It depends on the meter type defined. The meter type can be bytes or packets. This field by itself has no units. Please refer to https://github.com/opencomputeproject/SAI/blob/16c11ee7062e630f1eaf4bb3e44d31ab58eafa5b/inc/saipolicer.h#L123

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 (hunk 4 above)
Comment (dgsudharsan) [reply src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:64]: Please refer to my earlier comments regarding the units.

Comment (ganglyu) [reply src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:64]: Can you update the descriptions for CIR, CBS, PIR and PBS to add unit?

File: src/sonic-yang-models/yang-templates/sonic-policer.yang.j2 (hunk 2 above)
Comment (ganglyu) [reply src/sonic-yang-models/yang-templates/sonic-policer.yang.j2:57]: > @ganglyu Please refer to copp yang table. I have reused the same definitions used there. Basically copp internally uses policer and both should be aligned in yang definitions and so I defined as in https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-copp.yang

Can you update the descriptions for CIR, PIR, CBS and PBS to add unit?

**Comment by ganglyu** [comment]
> @ganglyu Please refer to copp yang table. I have reused the same definitions used there. Basically copp internally uses policer and both should be aligned in yang definitions and so I defined as in https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-copp.yang

Can you update the descriptions for CIR, PIR, CBS and PBS to add unit?

File: src/sonic-yang-models/tests/yang_model_tests/tests/policer.json (hunk 1 above)
Comment (dgsudharsan) [reply src/sonic-yang-models/tests/yang_model_tests/tests/policer.json]: Done

**Comment by dgsudharsan** [comment]
> > @ganglyu Please refer to copp yang table. I have reused the same definitions used there. Basically copp internally uses policer and both should be aligned in yang definitions and so I defined as in https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-copp.yang
> 
> Can you update the descriptions for CIR, PIR, CBS and PBS to add unit?

Done

**Review by ganglyu — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@lguohan could you please signoff?

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**